src/utils/SocketClient.hxx

```diff
@@ -71,17 +84,21 @@ public:
      *                      connecting on error.
      * @param timeout_seconds time in seconds that the connect is supposed to
      *                        timeout and look for a possible shutdown.
+     * @param status callback method for status as the connection attempt
+     *               progresses
```

Comment:
Note that the callback will be invoked on a different thread than the service executor.